packages/admin/src/pages/build-application/[applicationId]/[sectionId]/question-type/index.getServerSideProps.tsx

```diff
@@ -46,78 +87,125 @@ function getRadio(responseType: ResponseType) {
 export const getServerSideProps = async (
   context: GetServerSidePropsContext
 ) => {
-  const { params, req } = context;
+  const { params, req, query } = context;
   const { applicationId, sectionId } = params as Record<string, string>;
+  const { questionId, backTo } = query;
+  const queryString = buildQueryStringWithoutUndefinedValues({ backTo });
 
   const sessionId = getSessionIdFromCookies(req);
 
   const handleRequest = async (body: RequestBody) => {
     const { _csrf, ...props } = body;
 
-    if (redirectQuestionType.includes(body.responseType)) {
-      await addFieldsToSession('newQuestion', props, sessionId);
-      return {
-        redirectQuestionType: body.responseType,
-      };
-    }
-
     const questionSummary = (await getSummaryFromSession(
       'newQuestion',
       sessionId
     )) as QuestionSummary;
     const { optional, ...restOfQuestionSummary } = questionSummary;
-    const maxWords =
-      body.responseType === ResponseType.ShortAnswer
-        ? SHORT_QUESTION_WORD_LIMIT
-        : undefined;
-
-    await postQuestion(sessionId, applicationId, sectionId, {
-      ...restOfQuestionSummary,
-      ...props,
-      validation: {
-        maxWords,
-        mandatory: optional !== 'true',
-      },
-    });
 
-    return {
-      data: 'QUESTION_SAVED',
-      sessionId,
-    };
+    if (questionId) {
+      if (redirectQuestionType.includes(body.responseType)) {
+        await addFieldsToSession('updatedQuestion', props, sessionId);
+        return {
+          redirectQuestionType: body.responseType,
+        };
+      }
+
+      await patchQuestion(
+        sessionId,
+        applicationId,
+        sectionId,
+        questionId.toString(),
+        {
+          ...body,
+          ...props,
+          validation: {
+            mandatory: optional !== 'true',
+            maxWords: '',
+          },
+        }
+      );
+
+      return {
+        data: 'QUESTION_UPDATED',
+        sessionId,
+      };
+    } else {
+      if (redirectQuestionType.includes(body.responseType)) {
```

Comment:
We do this check on both sides of the if conditional, can we move it above the check for questionId?